### ISSUE TITLE
Plots behave glitchy.

### ISSUE THREAD

**Issue by mcourteaux** [wolfpld/tracy#906]
See the video:
[Wayfarer_2024-10-08_112613.webm](https://github.com/user-attachments/assets/1f9ca6c8-e3f3-45df-be7a-08210474e3e5)

I'm currently on e6eb9899

**Comment by wolfpld** [comment]
> Zooming reveals that the data is actually completely bogus. See the video:

Yes, this looks to be completely broken. Can you share a trace with these plots?

**Comment by mcourteaux** [comment]
[glitchy_plots.tracy.zip](https://github.com/user-attachments/files/17291502/glitchy_plots.tracy.zip)

I had to zip again, to make Github accept it.

**Comment by wolfpld** [comment]
I can reproduce the issue with your trace. It's certainly something interesting.

![Zrzut ekranu_20241008_180615](https://github.com/user-attachments/assets/8ee923f0-55fc-42df-aca5-b3d26d299067)

Without looking at the data, this looks like an unsorted plot, as you can see values going forward and backward in the screen below, and the glitching you see when zooming is a result of a binary search that can't work properly.

Tracy handles out-of-order plot points by design, this is a very old functionality and there were no problems with it before. I am not able to reproduce your issue with the `test/` application that generates millions of plot points from two threads, which results in a lot of resynchronization. The glitchy plots in your trace are in the 50-500 plot points range, and I'd expect there's no need to synchronize anything at all, with such low numbers.

The only recent thing that may affect this that comes to mind is replacement of the parallel STL with the ppqsort library.

**Comment by mcourteaux** [comment]
Is this something we can test by going back to before ppqsort got introduced _with_ the same trace file? I'm assuming the ppqsort stuff happens in the Tracy UI, but perhaps the trace file format changed since then? If it's not, I can attempt my first-ever git-bisect. 😄

**Comment by wolfpld** [comment]
PPQSort was added in 1c1faeff2. The trace format has not changed. Opening the existing trace will not result in any changes, you would have to redo the trace and see what comes out.

**Comment by lefp** [comment]
I haven't encountered the bug since that commit (e300d56f68ca08ca99ace9ce4bfd947073b97140). Thanks!
